services/loggers/class-wc-ebanx-debug-logger.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+use EBANX\Plugin\Services\WC_EBANX_Log;
+
+/**
+ * Log Checkout event data
+ */
+final class WC_EBANX_Debug_Logger extends WC_EBANX_Logger {
+	/**
+	 *
+	 * @param array $log_data data to be logged.
+	 * @param string $event event name to be logged.
+	 */
+	public static function persist( array $log_data = [], $event = null ) {
+		if (empty($event)) {
+			throw new BadMethodCallException('Event name must not be null on debug logs.');
```

Comment:
The parameter is null by default, but it cannot be null?

Reply:
Because of the signature. All methods MUST HAVE the exact signature

Reply:
This is what happens when a derivated method change the signature:

```
PHP Fatal error:  Declaration of SampleB::method() must be compatible with SampleA::method($event = null)
```

OR

```
PHP Fatal error:  Declaration of SampleB::method(string $event) must be compatible with SampleA::method($event = null)
```